sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -1301,6 +1355,287 @@ block_load(AccessorT acc, simd_mask<1> pred, PropertyListT props = {}) {
   return block_load<T, N>(acc, 0, pred, PassThru, props);
 }
 
+/// Each of the following block store functions stores a contiguous memory block
+/// to the address referenced by the USM pointer 'ptr', or from 'ptr +
+/// offset', where 'offset' is the offset in bytes (not in elements!) with data
+/// specified by 'vals'.
+/// The parameter 'pred' is the one element predicate. If it is set to 1, then
+/// all 'N' elements are stored. Otherwise, the block store operation is a
+/// NO-OP. The parameter 'props' specifies the optional compile-time properties
+/// of the type esimd::properties and may include esimd::cache_hint_L1,
+/// esimd::cache_hint_L2, esimd::cache_hint_L3, esimd::alignment.
+///
+/// void block_store(T* ptr, simd<T, N> vals, props={}); // (1)
+/// void block_store(T* ptr, size_t byte_offset,         // (2)
+///                          simd<T, N> vals, props={});
+
+/// void block_store(T* ptr, simd<T, N> vals,            // (3)
+///             simd_mask<1> pred, props={});
+
+/// void block_store(T* ptr, size_t byte_offset,         // (4)
+/// simd<T, N> vals, simd_mask<1> pred, props={});
+///
+/// void block_store(T* ptr, simd<T, N> vals, props={}); // (1)
+/// This function stores a contiguous memory block to USM pointer \p ptr
+/// with data specified by \p vals.
+///
+/// There may be temporary restrictions depending on L1, L2 cache hints,
+/// See details in the 'Restrictions' section below. The restrictions will be
+/// relaxed in the future.
+///
+/// The parameter \p props specifies the optional compile-time properties
+/// of the type esimd::properties and may include esimd::cache_hint_L1,
+/// esimd::cache_hint_L2, esimd::alignment. Other properties are ignored.
+///
+/// Cache hints: If \p props does not specify any L1 or L2 cache hints, then
+/// the cache_hint::none value is assumed by default.
+///
+/// Alignment: If \p props does not specify the 'alignment' property, then
+/// the default assumed alignment is the minimally required element-size
+/// alignment. Note that additional/temporary restrictions may apply
+/// (see Restrictions below).
```

Comment:
Major concern here: if apply the statements here, then the new block_store(usm_ptr, value) call may work slower than the old one because on Gen12 STORE requires 16-bytes alignment to produce block_store operation. Otherwise, scatter will be generated.

Assuming 4-byte alignment for store of simd<int, N> is valid, but will be a regression.
I think we need to raise the expected alignment to 16-bytes if cache-hints are not passed and predicate is not used. That will be consistent with the old block_store that had the default alignment = overaligned_tag<detail::OperandSize::OWORD>

Reply:
I believe I implemented this in my latest commit and updated comments as necessary, but please review carefully.